feat(icons): new icons dec 2024 - jan 2025

DanaPaslaru opened the pull request:
Submitting new UI icons from December 2024 through January 2025 all additions and updates approved by BXD

18 new icons
updated categories.yml with additions and updates
updated icons.yml with additions

updated ui-icon-master.ai file

#### Changelog

**New**

- 32/array.svg
- 32/asset-movement.svg
- 32/avro.svg
- 32/battery--error.svg
- 32/battery--warning.svg
- 32/channels.svg
- 32/create-link.svg
- 32/delivery-settings.svg
- 32/ibm-consulting-advantage--agent.svg
- 32/ibm-consulting-advantage--application.svg
- 32/ibm-consulting-advantage--assistant.svg
- 32/ibm--databand.svg
- 32/kiosk-device.svg
- 32/logo--git.svg
- 32/logo--red-hat-ai-instructlab-on-ibm-cloud.svg
- 32/multiuser-device.svg
- 32/spyre-accelerator.svg
- 32/trigger.svg

DanaPaslaru committed:
feat(icons): new icons dec 2024 - jan 2025

Submitting new UI icons from December 2024 through January 2025 all additions and updates approved by BXD

12 new icons

updated categories.yml with additions and updates
updated icons.yml with additions

DanaPaslaru committed:
Update ui-icon-master.ai

updated ui icon master file .ai

laurenmrice reviewed (approved):
✅

DanaPaslaru committed:
added 6 new ui icons

added 6 new icons
updated categories.yml with additions and updates
updated icons.yml with additions and updates

updated ui-icon-master.ai file

emyarod reviewed (approved):
looks good to me!

emyarod committed:
Merge branch 'main' into icons-new-dec-2024-jan-2025

emyarod commented on packages/icons/icons.yml:
```diff
@@ -1586,6 +1595,16 @@
     - asset
   sizes:
     - 32
+- name: asset--movement
```
```suggestion
- name: asset-movement
```

it looks like this extra dash is what's causing the test failures

DanaPaslaru committed:
Update icons.yml

updated icons.yml with suggested change:
from name "asset--movement" to "asset-movement"

kennylam committed:
Merge branch 'main' into icons-new-dec-2024-jan-2025

kennylam committed:
chore(jest): update snapshots

kennylam committed:
Merge branch 'main' of github.com:carbon-design-system/carbon into pr/DanaPaslaru/18388
